module/zfs/vdev.c

```diff
@@ -2689,42 +2689,79 @@ vdev_accessible(vdev_t *vd, zio_t *zio)
  * Get statistics for the given vdev.
  */
 void
-vdev_get_stats(vdev_t *vd, vdev_stat_t *vs)
+__vdev_get_stats(vdev_t *vd, vdev_stat_t *vs)
```

Comment:
"__" -> "impl": will fix.

Correct, it's done in the leaves.  The major reason for the change was to make it easy to aggregate histogram numbers to the top level vdevs/pools.  IIRC, before my patch the mid level vdevs, like raidz, wouldn't aggregate the histo numbers from their children correctly, since it was only aggregating from the direct children underneath a root vdev.